linux/net/rina/dt-utils.c

```diff
@@ -867,3 +890,47 @@ int rtxq_nack(struct rtxq * q,
 
         return 0;
 }
+
+int common_efcp_pdu_send(struct efcp * efcp,
+        		 struct rmt *  rmt,
+        	         address_t     address,
+                         qos_id_t      qos_id,
+			 struct pdu *  pdu)
+{
+        const struct pci *	pci;
+	struct efcp_container * efcpc;
+	cep_id_t		dest_cep_id;
+
+	if (!pdu)
+	        return -1;
+
+	pci = pdu_pci_get_ro(pdu);
+	if (!pci)
+		return -1;
+
+	/* Remote flow case */
+	if (pci_source(pci) != pci_destination(pci)) {
+	        if (rmt_send(rmt, address, qos_id, pdu)) {
+	                LOG_ERR("Problems sending PDU to RMT");
+	                return -1;
+	        }
+	        return 0;
+	}
+
+	/* Local flow case */
```

Comment:
Where are the separate if constructs?

checking for errors in the kernel code (at least our code) is alwaus If (!whateve) return error;

There is just a single if, with a return inside.